### PR TITLE
Do not refine mip/minip if nothing was found

### LINKED ISSUE #2114: `VolumeVisual` max/min intensity projection failure mode
The `VolumeVisual` has an interesting failure mode in both maximum and minimum intensity projection modes highlighted by @brisvag in the following [comment](https://github.com/napari/napari/pull/2934/files/c5c96e2b5faa854600f3d65d68a2ae01f47d8c7a#r660050470) which includes a nice video

If real maxima/minima are not found when the rays are cast, the value is searched for around `start_loc` of the ray anyway, causing the front faces of the cuboid to include some artefacts.

### PR DESCRIPTION
**Pull request by brisvag**
This fixes #2114. While this does not affect anything currently, it will cause problems in the future (such as with clipping planes).

### CODE REVIEW

**Review by alisterburt — APPROVED** [review]
good stuff, works perfectly!

**Review by almarklein — APPROVED** [review]
Thanks!  How do you mean that this change will affect clipping planes in the future? I guess we'll deal with it then?

File: vispy/visuals/volume.py [hunk 1]
```diff
@@ -335,13 +337,15 @@
...
     after_loop="""
-        // Refine search for min value
-        loc = start_loc + step * (float(mini) - 0.5);
-        for (int i=0; i<10; i++) {
-            minval = min(minval, $sample(u_volumetex, loc).r);
-            loc += step * 0.1;
-        }
-        gl_FragColor = applyColormap(minval);
+        // Refine search for min value, but only if anything was found
+        if ( maxi > -1 ) {{
```
[the remark this answers is not in the document]
Comment (brisvag) [reply vispy/visuals/volume.py:341]: whoops! The dangers of copypasting :)

**Comment by brisvag** [comment]
> Thanks! How do you mean that this change will affect clipping planes in the future? I guess we'll deal with it then?

This change will affect them in the sense that it preemptively solves a bug that was uncovered by a tentative implementation of clipping planes (see [this comment](https://github.com/napari/napari/pull/2934#discussion_r660050470))

**Comment by djhoese** [comment]
Looks good. Thanks for porting this so quickly.

**Pull request merged**